### PR TITLE
show pdf preview in the files sidebar

### PR DESCRIPTION
**Pull request by icewind1991**
- [x] requires https://github.com/nextcloud/server/pull/2435

Before:

After:

cc @jancborchardt @MorrisJobke

### CODE REVIEW

File: js/previewplugin.js [hunk 1]
```diff
@@ -146,3 +146,43 @@ $(document).ready(function(){
 		}
 	}
 });
+
+(function () {
+
+	var SidebarPreview = function () {
+		this.preview = new OCA.Files_Markdown.Preview();
```
Comment (nickvergessen) [inline js/previewplugin.js]: Inter app dependency or mistake?

Comment (icewind1991) [inline js/previewplugin.js]: I would never just copy-past a block of code and forget to properly clean things up

**Comment by icewind1991** [comment]
@nickvergessen fixed

**Comment by jancborchardt** [comment]
Similar here:
```
TypeError: this.preview is undefined in previewplugin.js:153:3
```

Btw when this works, we have another widely used file type ready for a possible grid view, right?

**Comment by LukasReschke** [comment]
> Btw when this works, we have another widely used file type ready for a possible grid view, right?

No. I doubt that this will scale at all. We need a server-side preview generation for that.

**Comment by MorrisJobke** [comment]
Ummm what? We simply load the PDF in the side bar? :)


This will not work for a grid view in my opinion. At least once you have something like 10 PDFs in a folder it will get tricky.

**Comment by icewind1991** [comment]
Fixed the error

I agree that this should be used for grid

**Comment by MorrisJobke** [comment]
> I agree that this should be used for grid

I guess there is a `not` missing ;)

**Comment by MorrisJobke** [comment]
Tested and works 👍 but we should not use this in a grid view.

**Comment by icewind1991** [comment]
> I guess there is a not missing ;)

Correct

**Comment by jancborchardt** [comment]
Okidoke, thanks for clarification :)

**Pull request merged**